### PR TITLE
Change error messages in pre-built functions to make the descriptions more accurate

### PR DESCRIPTION
**Pull request by cosmicshuai**
fixes: #3955 
replace $"{expr} can't evaluate." with more detailed messages based on the context of some pre-built functions

### CODE REVIEW

File: libraries/AdaptiveExpressions/ExpressionFunctions.cs [hunk 1]
```diff
@@ -772,7 +772,7 @@ public static ExpressionEvaluator TimeTransform(string type, Func<DateTime, int,
                         }
                         else
                         {
-                            error = $"{expr} could not be evaluated";
+                            error = $"{expr} should contain an ISO format timestamp and an integer of time interval.";
```
Comment (chrimc62) [inline libraries/AdaptiveExpressions/ExpressionFunctions.cs:775]: >of [](start = 99, length = 2)

I would drop this word.

File: libraries/AdaptiveExpressions/ExpressionFunctions.cs [hunk 2]
```diff
@@ -3483,7 +3483,7 @@ private static IEnumerable<object> Flatten(IEnumerable<object> list, int dept)
                             }
                             else
                             {
-                                error = $"{expr} can't evaluate.";
+                                error = $"{expr} should contain an ISO format timestamp, an integer of time interval, a string represents unit of time and an optional string of output format.";
```
Comment (chrimc62) [inline libraries/AdaptiveExpressions/ExpressionFunctions.cs:3486]: >should contain an ISO format timestamp, an integer of time interval, a string represents unit of time and an optional string of output format [](start = 49, length = 141)

This is awkward.  How about "should contain an ISO format timestamp, a time interval integer, a string unit of time and an optional output format string."

File: libraries/AdaptiveExpressions/ExpressionFunctions.cs [hunk 3]
```diff
@@ -3578,7 +3578,7 @@ private static IEnumerable<object> Flatten(IEnumerable<object> list, int dept)
                             }
                             else
                             {
-                                error = $"{expr} can't evaluate.";
+                                error = $"{expr} should contain an integer of time interval, a string represents unit of time and an optional string of output format.";
```
Comment (chrimc62) [inline libraries/AdaptiveExpressions/ExpressionFunctions.cs:3581]: >should contain an integer of time interval, a string represents unit of time and an optional string of output format. [](start = 49, length = 117)

Same comment for the other strings.

File: libraries/AdaptiveExpressions/ExpressionFunctions.cs [hunk 4]
```diff
@@ -3945,7 +3945,7 @@ private static IEnumerable<object> Flatten(IEnumerable<object> list, int dept)
                             }
                             else
                             {
-                                error = $"{expr} can't evaluate.";
+                                error = $"{expr} should contain a string of URI.";
```
Comment (chrimc62) [inline libraries/AdaptiveExpressions/ExpressionFunctions.cs:3948]: >string of URI [](start = 66, length = 13)

URI string

File: libraries/AdaptiveExpressions/ExpressionFunctions.cs [hunk 5]
```diff
@@ -3969,7 +3969,7 @@ private static IEnumerable<object> Flatten(IEnumerable<object> list, int dept)
...
```
Comment (chrimc62) [inline libraries/AdaptiveExpressions/ExpressionFunctions.cs:3972]: >string of URI [](start = 66, length = 13)

URI string.

File: libraries/AdaptiveExpressions/ExpressionFunctions.cs [hunk 6]
```diff
@@ -3993,7 +3993,7 @@ private static IEnumerable<object> Flatten(IEnumerable<object> list, int dept)
...
```
Comment (chrimc62) [inline libraries/AdaptiveExpressions/ExpressionFunctions.cs:3996]: > string of URI [](start = 65, length = 14)

Same for all of these.

File: libraries/AdaptiveExpressions/ExpressionFunctions.cs (hunk 1 above)
Comment (cosmicshuai) [reply libraries/AdaptiveExpressions/ExpressionFunctions.cs:775]: changed to $"{expr} should contain an ISO format timestamp and and a time interval integer."

File: libraries/AdaptiveExpressions/ExpressionFunctions.cs (hunk 2 above)
Comment (cosmicshuai) [reply libraries/AdaptiveExpressions/ExpressionFunctions.cs:3486]: changed

File: libraries/AdaptiveExpressions/ExpressionFunctions.cs (hunk 3 above)
Comment (cosmicshuai) [reply libraries/AdaptiveExpressions/ExpressionFunctions.cs:3581]: changed

File: libraries/AdaptiveExpressions/ExpressionFunctions.cs (hunk 4 above)
Comment (cosmicshuai) [reply libraries/AdaptiveExpressions/ExpressionFunctions.cs]: changed to  should contain a URI string.
Also applied to the below ones

File: libraries/AdaptiveExpressions/ExpressionFunctions.cs [hunk 7]
```diff
@@ -3607,7 +3607,7 @@ private static IEnumerable<object> Flatten(IEnumerable<object> list, int dept)
                             }
                             else
                             {
-                                error = $"{expr} can't evaluate.";
+                                error = $"{expr} should a time interval integer, a string unit of time and an optional output format string.";
```
[the remark this answers is not in the document]
Comment (cosmicshuai) [reply libraries/AdaptiveExpressions/ExpressionFunctions.cs]: fixed

**Pull request merged**